examples/01_plotting/show_ball_sampling.py

```diff
@@ -0,0 +1,74 @@
+"""
+Illustration of the volume to surface sampling scheme
+=====================================================
+
```

Comment:
Yes if we keep it this example should be reworked. Originally, I had only thrown
it in there to illustrate the sampling method for reviewers by producing the 2d
plot at the top of the PR, and was planning to remove it. However @GaelVaroquaux
suggested that it might be interesting to users as well. The only problem is
that in that case we need a coord_transform that can handle 2d images, which is
why I'm not using nilearn.image.coord_transform for now. It's a one-line
function though.
If we keep it I will also add an illustration for the line sampling.